warehouse/oidc/models/_core.py

```diff
@@ -189,10 +190,12 @@ def verify_claims(self, signed_claims: SignedClaims):
                         f"JWT for {self.__class__.__name__} is missing claim: "
                         f"{claim_name}"
                     )
-                return False
+                raise InvalidPublisherError(f"Missing claim {claim_name!r}")
 
             if not check(getattr(self, claim_name), signed_claim, signed_claims):
```

Comment:
Nonblocker: we could probably push the context even further here by having each `check` also raise rather than return; just a thought in case we need to in the future 🙂

Reply:
I am reserving that as an option if we need it, but didn't want to have to write a bunch of new claim-specific tests if I didn't have to. 🙂